backend/modules/metadata_store/prismastore.py

```diff
@@ -255,18 +257,19 @@ async def aassociate_data_source_with_collection(
             if existing_collection_associated_data_sources:
                 existing_collection_associated_data_sources[
                     data_src_to_associate.data_source_fqn
-                ] = data_src_to_associate.dict()
+                ] = data_src_to_associate
             else:
                 existing_collection_associated_data_sources = {
-                    data_src_to_associate.data_source_fqn: data_src_to_associate.dict()
+                    data_src_to_associate.data_source_fqn: data_src_to_associate
                 }
 
+            logger.info(existing_collection_associated_data_sources)
             associated_data_sources: Dict[str, Dict[str, Any]] = {}
             for (
                 data_source_fqn,
                 data_source,
             ) in existing_collection_associated_data_sources.items():
-                associated_data_sources[data_source_fqn] = data_source.model_dump()
+                associated_data_sources[data_source_fqn] = data_source.serialize()
```

Comment:
Why serialize() and not model_dump() ?

Also what if some item is already a dict?